src/main/cms/cms_menu_firmware.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//
+// Firmware related menu contents and support functions
+//
+
+#include <stdbool.h>
+
+#include "platform.h"
+
+#ifdef USE_CMS
+
+#include "build/version.h"
+
+#include "cms/cms.h"
+#include "cms/cms_types.h"
+
+#include "config/config.h"
+
+#include "drivers/system.h"
+
+#include "sensors/acceleration.h"
+#include "sensors/barometer.h"
+#include "sensors/gyro.h"
+
+#include "cms_menu_firmware.h"
+
+
+// Calibration
+
+static long cmsCalibrateGyro(displayPort_t *pDisp, const void *self)
+{
+    UNUSED(pDisp);
+    UNUSED(self);
+
+    gyroStartCalibration(false);
+
+    return 0;
+}
+
+#if defined(USE_ACC)
+static long cmsCalibrateAcc(displayPort_t *pDisp, const void *self)
+{
+    UNUSED(pDisp);
+    UNUSED(self);
+
+    accStartCalibration();
+
+    return 0;
+}
+#endif
+
+#if defined(USE_BARO)
+static long cmsCalibrateBaro(displayPort_t *pDisp, const void *self)
+{
+    UNUSED(pDisp);
+    UNUSED(self);
+
+    baroStartCalibration();
+
+    return 0;
+}
+#endif
+
+static const OSD_Entry menuCalibrationEntries[] = {
+    { "--- CALIBRATE ---", OME_Label, NULL, NULL, 0 },
+    { "GYRO", OME_Funcall, cmsCalibrateGyro, NULL, 0 },
+#if defined(USE_ACC)
+    { "ACC",  OME_Funcall, cmsCalibrateAcc, NULL, 0 },
+#endif
+#if defined(USE_BARO)
+    { "BARO", OME_Funcall, cmsCalibrateBaro, NULL, 0 },
+#endif
+    { "BACK", OME_Back, NULL, NULL, 0 },
+    { NULL, OME_END, NULL, NULL, 0 }
+};
```

Comment:
Why not have compass calibration here as well?

Reply:
I hummed and harred about that. Technically nothing wrong with doing it. The difference between gyro / ACC / BARO calibration and MAG calibration is that the former are 'hands off' calibrations, whereas the latter requires the user to move the craft by hand during the calibration cycle. To me, the use case of a user moving a craft with plugged in battery while wearing goggles seemed to be inherently unsafe, so I thought it was better to not support it.
But open to discussion.